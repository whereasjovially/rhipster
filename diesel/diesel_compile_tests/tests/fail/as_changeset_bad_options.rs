#[macro_use]
extern crate diesel;

table! {
    users {
        id -> Integer,
        name -> Text,
    }
}

#[derive(AsChangeset)]
#[table_name = "users"]
#[changeset_options(treat_none_as_null("true"))]
struct UserForm1 {
    id: i32,
    name: String,
}

#[derive(AsChangeset)]
#[table_name = "users"]
#[changeset_options(treat_none_as_null)]
struct UserForm2 {
    id: i32,
    name: String,
}

#[derive(AsChangeset)]
#[table_name = "users"]
#[changeset_options(treat_none_as_null = "foo")]
struct UserForm3 {
    id: i32,
    name: String,
}

#[derive(AsChangeset)]
#[table_name = "users"]
#[changeset_options()]
struct UserForm4 {
    id: i32,
    name: String,
}

#[derive(AsChangeset)]
#[table_name = "users"]
#[changeset_options = "treat_none_as_null"]
struct UserForm5 {
    id: i32,
    name: String,
}

#[derive(AsChangeset)]
#[table_name = "users"]
#[changeset_options]
struct UserForm6 {
    id: i32,
    name: String,
}

fn main() {}
