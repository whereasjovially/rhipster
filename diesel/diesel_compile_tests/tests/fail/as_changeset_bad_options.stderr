error: `treat_none_as_null` must be in the form `treat_none_as_null = "true"`
  --> $DIR/as_changeset_bad_options.rs:13:21
   |
13 | #[changeset_options(treat_none_as_null("true"))]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `treat_none_as_null` must be in the form `treat_none_as_null = "true"`
  --> $DIR/as_changeset_bad_options.rs:21:21
   |
21 | #[changeset_options(treat_none_as_null)]
   |                     ^^^^^^^^^^^^^^^^^^

error: `treat_none_as_null` must be in the form `treat_none_as_null = "true"`
  --> $DIR/as_changeset_bad_options.rs:29:21
   |
29 | #[changeset_options(treat_none_as_null = "foo")]
   |                     ^^^^^^^^^^^^^^^^^^^^^^^^^^

error: Missing required option `treat_none_as_null`
  --> $DIR/as_changeset_bad_options.rs:37:3
   |
37 | #[changeset_options()]
   |   ^^^^^^^^^^^^^^^^^^^

error: `changeset_options` must be in the form `changeset_options(...)`
  --> $DIR/as_changeset_bad_options.rs:45:3
   |
45 | #[changeset_options = "treat_none_as_null"]
   |   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: `changeset_options` must be in the form `changeset_options(...)`
  --> $DIR/as_changeset_bad_options.rs:53:3
   |
53 | #[changeset_options]
   |   ^^^^^^^^^^^^^^^^^
